gpt_index/indices/base.py

```diff
@@ -200,8 +200,17 @@ def _get_nodes_from_document(
         for i, text_chunk in enumerate(text_chunks):
             fmt_text_chunk = truncate_text(text_chunk, 50)
             print(f"> Adding chunk: {fmt_text_chunk}")
+            # calculate the overlap between the current and previous chunk
+            STRING_OVERLAP_TEST_SIZE = 20  # chosen arbitrarily
```

Comment:
if the max_chunk_overlap is like 1 this won't work right? since all 20 characters of the string need to match for this to be valid

an alternative/maybe cleaner idea is to modify the text_splitter.split_text, that can return an overall "TextSplit" object: 
```
@dataclass
class TextSplit:
   text_chunk: str
   num_char_overlap: int
```
num_char_overlap can be computed during text splitting time since there is logic in `split_text` to get the text overlap given the `max_chunk_overlap` param. I think this is cleaner than re-attempting to compute the overlap post-hoc.

Since `split_text` inherits from langchain you may not be able to directly change the return signature, you can maybe define a separate method like `split_text_and_get_object` (idk). And then `split_text` could just call this method under the hood (and only return the strings)